nntrainer/compiler/tflite_interpreter.cpp

```diff
@@ -422,12 +415,91 @@ buildOperatorCodes(const TfOpIdxMap &map, flatbuffers::FlatBufferBuilder &fbb) {
   return fbb.CreateVector(fb_op_codes);
 }
 
+/**
+ * @brief get Serialized Tensor shape
+ *
+ * @param dim Tensor dimension to convert
+ * @param type tflite Tensor type
+ * @return flatbuffers::Offset<flatbuffers::Vector<int32_t>>
+ */
+flatbuffers::Offset<flatbuffers::Vector<int32_t>>
+buildTesorShape(const TensorDim &dim, flatbuffers::FlatBufferBuilder &fbb,
+                int effective_dimension, bool is_signature = false) {
+  std::vector<int32_t> fb_dimension;
+
+  /// batch dimension is always considered with signature
+  if ((effective_dimension >> 3) & 1) {
```

Comment:
Could you add more explanation about effective_dimension?

Reply:
We have fixed number of dimension, to support flexible dimension. we need to store the squeezed version of dimension somewhere. 

for example(3, 1, 1, 5) can be (3, 5), (3, 1, 5), (3, 1, 1, 5). `effective_dimension` distinguish between those.

Reply:
can you add as comments the possible values of effective_dimension? 
also what does >> 3 signify - just checking the batch value?

Reply:
Yes, 3 signifies checking the batch value that will be always added in our case.
Added as a doxygen, this is changed to std::bitset<N> to make it explicit in few pr's ahead.

Reply:
this is fine for now, but later, add a method like `getBatchIndex` or `getBatchLocation` in `TensorDim`, and right shift by that value?

Reply:
Yes, Regarding this, I have made some (arguable) changes regarding this in the upcoming pr :)

Reply:
cool.

Reply:
if (3,5) and effective dimension is 1, 0, 0, 1 .. then it is same with (3, 1,  1, 5)?? or.. it the dimension is (3, 1,1,5), and effective dimension is 0, 1,1, 0?? is it wrong? need to check?

Reply:
If our dimension is 3, 1, 1, 5 and 1, 0, 0, 1. It means (3, 5) in flatbuffer schema

If our dimension is (3, 1, 1, 5) and (0, 1, 1, 0). This would be an invalid combination but interpreted as (1, 1) in flatbuffer schema

Reply:
So.. simply it ignore the dimension which is zero?

Reply:
Yes. I guess we can decide to keep dimension if it is not one in the future.